ios/api.md

```diff
@@ -467,6 +467,166 @@ do {
+
+    class func getAppSyncClient(mode: AppSyncClientMode) -> AWSAppSyncClient? {
+        return clients[mode];
+    }
+
+    class func initClients() throws {
+        let serviceConfigAPIKey = try AWSAppSyncServiceConfig()
+        let cacheConfigAPIKey = try AWSAppSyncCacheConfiguration(useClientDatabasePrefix: true,
+                                                                    appSyncServiceConfig: serviceConfigAPIKey)
+        let clientConfigAPIKey = try AWSAppSyncClientConfiguration(appSyncServiceConfig: serviceConfigAPIKey,
+                                                                 cacheConfiguration: cacheConfigAPIKey)
+        clients[AppSyncClientMode.public] = try AWSAppSyncClient(appSyncConfig: clientConfigAPIKey)
+
+        let serviceConfigIAM = try AWSAppSyncServiceConfig(forKey: "friendly_name_AWS_IAM")
+        let cacheConfigIAM = try AWSAppSyncCacheConfiguration(useClientDatabasePrefix: true,
+                                                                 appSyncServiceConfig: serviceConfigIAM)
+        let clientConfigIAM = try AWSAppSyncClientConfiguration(appSyncServiceConfig: serviceConfigIAM,
+                                                                  cacheConfiguration: cacheConfigIAM)
+        clients[AppSyncClientMode.private] = try AWSAppSyncClient(appSyncConfig: clientConfigIAM)
+    }
+}
+```
+
+This is what the usage would look like.
+
+```swift
+ClientFactory.getAppSyncClient(AppSyncClientMode.private)?.fetch(query: ListPostsQuery(), cachePolicy: .returnCacheDataDontFetch)  { (result, error) in
+            if error != nil {
+                print(error?.localizedDescription ?? "")
+                return
+            }
+            self.postList = result?.data?.listPosts
+        };
+```
+
+The following example uses `API_KEY` as the default authorization mode and `AWS_IAM` as an additional authorization mode.
+
+```
+type Post @aws_api_key
+          @aws_iam {
+	id: ID!
+	author: String!
+	title: String
+	content: String
+	url: String @aws_iam
+	ups: Int @aws_iam
+	downs: Int @aws_iam
+	version: Int!
+}
+```
+
+1. Add a post (Mutation) through `ClientFactory.getAppSyncClient(AppSyncClientMode.private)` using `AWS_IAM` authorization mode.
+
+2. Query the post through `ClientFactory.getAppSyncClient(AppSyncClientMode.private)` using `AWS_IAM ` authorization mode.
+
+3. Query the post through `ClientFactory.getAppSyncClient(AppSyncClientMode.public)` using `API_KEY` authorization mode
+
+```swift
+appSyncClient?.fetch(query: GetPostQuery())  { (result, error) in
+    if error != nil {
+        print(error?.localizedDescription ?? "")
+        return
+    }
+    var post = result?.data?.getPost?
+    post.id
+    post.author
+    post.title
+    post.content
+    post.version
+    post.url // Null - because it's not authorized
+    post.ups // Null - because it's not authorized
+    post.downs // Null - because it's not authorized
```

Comment:
An example of how to retrieve the error.
`result?.errors![0].message.contains("Not Authorized to access url on type Post")`

Reply:
Added.